### DISCUSSION TITLE
Linq Select DateTime returns local time

### DISCUSSION THREAD

**Discussion by f1nzer** [JasperFx/marten#1274] [Q&A] [answered]
```cs
class Data 
{ 
   public DateTime Date { get; set; }
}
```
Query: `session.Query<Data>().ToList()` returns data with DateTime in UTC format.
Query: `session.Query<Data>().Select(x => new { x.DateTime }).ToList()` returns data in local format.

DateTime is indexed via duplicated field and `DuplicatedFieldUseTimestampWithoutTimeZoneForDateTime = false`

**Comment by oskardudycz** [comment, marked as answer]
@f1nzer could you check https://www.npgsql.org/doc/types/datetime.html? It seems to be that it looks like default Npgsql behaviour. See also more in: https://github.com/JasperFx/marten/pull/1179

**Comment by tetious** [comment]
Are you saying it is converting the date to the local timezone in the second case? That might indeed be a bug.

I'd expect the DateTime Kind set to `Unspecified` in both cases if I'm understanding the mapping documentation in the npgsql docs. `Unspecified` can make conversions a bit surprising if you don't understand how it influences them. 

If `Kind == Unspecified` and you call `ToUniversalTime`, you will get a DateTime converted as if the source was in the **local timezone**. If you call `ToLocalTime`, you will get a DateTime converted as if the source was in **UTC**. 

In other words, either conversion method will always result in a converted value when Kind is Unspecified, which is probably not what you want in this case.

When you know the source is in UTC, you can make ToUniversalTime "safe" by calling DateTime.SpecifyKind to set Kind to Utc.
